lib/Common/Memory/HeapAllocator.h

```diff
@@ -130,6 +130,34 @@ struct HeapAllocator
     static HeapAllocator Instance;
     static HeapAllocator * GetNoMemProtectInstance();
 
+private:
+    HANDLE PrivateHeap;
+
+    void InitPrivateHeap()
+    {
+        if (this->PrivateHeap == nullptr)
+        {
+            this->PrivateHeap = HeapCreate(0, 0, 0); // no options, default initial size, no max size
+        }
+    }
+
+    void DestroyPrivateHeap()
+    {
+        if (this->PrivateHeap != nullptr)
+        {
+            BOOL success = HeapDestroy(this->PrivateHeap);
+            Assert(success);
+            this->PrivateHeap = nullptr;
+        }
+    }
+
+    HANDLE GetPrivateHeap()
+    {
+        InitPrivateHeap(); // will initialize PrivateHeap if not already initialized
+        return this->PrivateHeap;
```

Comment:
I don't think that will buy us much.  Most of the allocation that we use the heap for require multithread access (JIT bail record, recycler metadata, etc.)